src/pq_flash_index.cpp

```diff
@@ -1167,27 +1226,29 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
     std::vector<Neighbor> &full_retset = query_scratch->full_retset;
 
     uint32_t best_medoid = 0;
+    uint32_t best_shard = 0;
     float best_dist = (std::numeric_limits<float>::max)();
-    if (!use_filter)
```

Comment:
regarding to the searching parameter "use_filter", if the index is built for filter diskann, we do not expect users to search the index with "use_filter" == false; if the index is built for non-filter diskann, we do not expect users to search the index with "use_filter" == true. So we could remove this parameter or at least we need to do some checking in case uses mis-use the parameter.

Reply:
yes. separate PR